contrib/vistools/test/test_integration.py

```diff
@@ -0,0 +1,30 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not
+# use this file except in compliance with the License. A copy of the License
+# is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+import os
+import pytest
+from filecmp import dircmp
+
+from generate_graphs import generate
+
+CWD = os.path.dirname(os.path.realpath(__file__))
+
+BEAM_COMPARISONS = [(os.path.join(CWD, "resources", "beams.json"),
+                     os.path.join(CWD, "resources", "output"))]
+
+@pytest.mark.parametrize("beams, expected_output", BEAM_COMPARISONS)
+def test_beam_generation(beams, expected_output, tmpdir):
```

Comment:
I'm assuming all files under `contrib/vistools/test/resources/` were created by the tool, rather than manually. As this test adds quite a bit of clutter, will be difficult to maintain, unless we always create new test output with the given version of the code, in which case it is unclear what the test covers as it will be passed by definition, I'm wondering whether this test is necessary in the current form.

What you really want to test is whether the tool runs successfully on a given beam history. Why not test that after running `generate` the expected files exist under the output directory? This would avoid the need for committing and more importantly updating all the test outputs, but at the same time at least make sure that the expected outputs are generated. To test whether all the outputs render correctly in the browser one anyway does not get around actually opening them in the browser.

Reply:
It's not the best test, I agree, but I wanted to add something. In it's current form, it already does what you suggest, it generates the beams from the `resources/beams.json` file and only checks that the same files exist in the output as in `resources/output` (it doesn't look if the contents match).

I understand that we don't want to update this very often. The format of `beams.json` is very sockeye-version agnostic, and I don't expect it to change very often. If it does change, the code in `generate_graphs.py` will have to as well, since they are quite coupled. 

All in all, I just wanted to make the minimal modifications to the package so it works for sockeye. If you think the current test doesn't bring value then I'll just get rid of it.

Reply:
yeah, makes sense. So how about the modification to not actually include the files but change the test to just check for their existence?

Reply:
I emptied out the files in the `resources/output` directory. The test continues to work as it was only checking that the files in that directory are named the same as the output of the script.